File: 1170. Compare Strings by Frequency of the Smallest Character.py
class Solution:
    def numSmallerByFrequency(self, q: List[str], w: List[str]) -> List[int]:
        F = lambda s: s.count(min(s))
        fw = sorted([F(s) for s in w])
        return [len(fw)-bisect.bisect_right(fw, F(s)) for s in q]
        

File: 1170. Compare Strings by Frequency of the Smallest Character.cpp
class Solution {
public:
    
    vector<int> numSmallerByFrequency(vector<string>& q, vector<string>& w) {
        auto F = [](const string& s) {
            return count(s.begin(), s.end(), *min_element(s.begin(), s.end()));
        };
        vector<int> fw;
        for (auto& s : w) 
            fw.push_back(F(s));
        sort(fw.begin(), fw.end());
        vector<int> ans;
        for (auto& s : q) 
            ans.push_back(fw.size() - (upper_bound(fw.begin(), fw.end(), F(s)) - fw.begin()));
        
        return ans;
    }
};